testing.rb

```diff
@@ -0,0 +1,389 @@
+# p(1..3).my_none?(&proc{|num| num%2==0})
+# p[1,2,3,4,5].tap{|t| t.my_none?{|n| n%3}}
+# p [false, nil,false].my_none?                            #=>true
+# p [false, false, []].my_none?                            #=>false
+# p [true,[]].my_none?(String)                             #=>true
+# p [true,[]].my_none?(Numeric)                            #=>true
+# p ['',[]].my_none?(String)                               #=>false
+# p ['dog','cat'].my_none?(/x/)                            #=>true
+# p ['dog', 'cat'].my_none?(/d/)                           #=>false
+# p ['dog','car'].my_none?(5)                              #=>true
+# p [5,'dog','car'].my_none?(5)                            #=>false
+
+# conditions for count
+  # p ary = [3, 3, 3, 2, 4, 2]
+  # p ary.my_count #=> 4
+  # p names = %w(luigui mario boozer luffy zorro jinbei)
+  # p names.my_count
+  # p ary.my_count
+  # p ary.my_count(3)            #=> 2
+  # p ary.my_count{ |x| x%2==0 } #=> 3
+# 2nd part of my_count
+# p [1,2,3,4,5].my_count                                      #=>5
+# p (1..3).my_count                                           #=>3
+# p [1,2,3].my_count(1)                                       #=>1
+# p [1,2,3].my_count(&proc{|num|num%2==0})                    #=>1
+# p [1,2,3,4,5].tap{|t| t.my_count{|n| n%3}}                  #=>[1,2,3,4,5]
+
+# condition for my map
+# p (1..4).my_map{ |i| i*i }      #=> [1, 4, 9, 16]
+# p (1..4).my_map { "cat"  }   #=> ["cat", "cat", "cat", "cat"]
+# p [1,2,3].my_map(&proc{|x|x%2})
+# p [1,2,3].map(&proc{|x|x%2})
+
+#   2nd part of my_map
+# p [1,2,3].my_map{|x| x==2}                                  #=>[1] just here there is doubt
+# p [1,2,3].my_map(&proc{|x|x*2})                             #=>[2,4,6]
+# p (1..3).my_map(&proc{|num| num+1})                         #=>[2.3.4]
+# p [1,2,3].my_map(&proc{|x|x%2})                             #[1,0,1]
+
+# conditions for my_inject
+# Sum some numbers
+# p (5..10).my_inject(:+)                             #=> 45
+# Same using a block and inject
+# p (5..10).my_inject { |sum, n| sum +  n }            #=> 45
+# Multiply some numbers
+# p (5..10).my_inject(1, :*)                          #=> 151200
+# Same using a block
+# p (5..10).my_inject(2) { |product, n| product * n } #=> 151200
+# find the longest word
+# longest = %w{ cat sheep bear biggest}.my_inject do |memo, word|
+#  memo.length > word.length ? memo : word
+# end
+# p longest                                        #=> "sheep"
+
+# 2nd part of my input
+# p [1,2,3].my_inject(&proc{|total, num| total*num})          #=>6
+# p (1..3).my_inject(&proc{|total, num| total*num})           #=>6
+# p (1..3).my_inject(4) { |prod, n| prod * n }                #=>24
+# p [1,2,3].my_inject(:+)                                     #=>6
+# p (1..9).my_inject(:+)                                      #=>45
+# p [1,2,3].my_inject(4, :+)                                  #=>24
+# p (1..3).my_inject(4, :*)                                   #=>24
```

Comment:
Kindly remove all not used code or examples, you can create another file with the examples.